File: web/lib/app-version.ts
import packageJson from "@/package.json"

export const APP_VERSION = {
  version: process.env.NEXT_PUBLIC_APP_VERSION ?? packageJson.version,
  gitSha:
    process.env.NEXT_PUBLIC_GIT_SHA ??
    process.env.NEXT_PUBLIC_VERCEL_GIT_COMMIT_SHA ??
    null,
  buildTime: process.env.NEXT_PUBLIC_BUILD_TIME ?? null,
} as const

export const APP_VERSION_LABEL = `v${APP_VERSION.version}`

export function appVersionDetailLabel(): string {
  const details = [
    APP_VERSION_LABEL,
    APP_VERSION.gitSha ? APP_VERSION.gitSha.slice(0, 7) : null,
    APP_VERSION.buildTime,
  ].filter(Boolean)
  return details.join(" · ")
}

File: app/core/app_version.py
from __future__ import annotations

import os


APP_VERSION = "0.1.1"
APP_GIT_SHA = os.getenv("APP_GIT_SHA") or os.getenv("GIT_SHA")
APP_BUILD_TIME = os.getenv("APP_BUILD_TIME") or os.getenv("BUILD_TIME")


def backend_version_payload() -> dict[str, str | None]:
    return {
        "version": APP_VERSION,
        "git_sha": APP_GIT_SHA,
        "build_time": APP_BUILD_TIME,
    }
